devices/Sim7080/Sim7080G.cs

```diff
@@ -0,0 +1,301 @@
+
+        /// <summary>
+        /// Cellular Network Operator
+        /// </summary>
+        public string Operator { get; private set; } = "Unknown";
+
+        /// <summary>
+        /// The Public IP Address
+        /// </summary>
+        public string IPAddress { get; private set; } = "0.0.0.0";
+
+        /// <summary>
+        /// Cloud-2-Device subscription topic Name
+        /// </summary>
+        public string SubTopic { get; set; }
+
+        /// <summary>
+        /// Device-2-Cloud publication topic Name
+        /// </summary>
+        public string PubTopic { get; set; }
+
+        /// <summary>
+        /// Set the network system mode for wireless and cellular network communication service
+        /// </summary>
+        /// <param name="systemMode">The desired <see cref="SystemMode"/></param>
+        /// <param name="enableReporting">Report the network system mode information</param>
+        /// <param name="wait">The time to wait to switch system mode</param>
+        public void SetNetworkSystemMode(SystemMode systemMode = SystemMode.GSM, bool enableReporting = true, int wait = 5000)
+        {
+            SimController.SetSystemMode(_serialPort, systemMode, enableReporting);
+
+            Thread.Sleep(wait);
+        }
+
+        /// <summary>
+        /// Connect to the cellular network
+        /// </summary>
+        //// <param name="apn">Cellular network access point name</param>
+        public void NetworkConnect(string apn)
+        {
+            if (!_serialPort.IsOpen)
+            {
+                NetworkConnected = ConnectionStatus.Disconnected;
+                return;
+            }
+
+            var retryCount = 0;
+
+            do
+            {
+                retryCount++;
+
+                SimController.NetworkConnect(_serialPort, apn);
+
+            } while (NetworkConnected == ConnectionStatus.Disconnected && retryCount < Retry);
+        }
+
+        /// <summary>
+        /// Disconnect to the cellular network 
+        /// </summary>
+        public void NetworkDisconnect()
+        {
+            if (_serialPort.IsOpen &&
+                NetworkConnected == ConnectionStatus.Connected)
+            {
+                SimController.NetworkDisconnect(_serialPort);
+            }
+        }
+
+        /// <summary>
+        /// Connect to Azure IoT Hub using MQTT protocol
+        /// </summary>
+        /// <param name="deviceId">The device identifier</param>
+        /// <param name="hubName">The Azure IoT Hub Name</param>
+        /// <param name="sasToken">The Secure Access Token</param>
+        /// <param name="portNumber">The MQTT port number</param>
+        /// <param name="apiVersion">The Azure IoT Hub API version</param>
+        /// <param name="wait">The time to wait to establish the connection</param>
+        public void ConnectAzureIoTHub(string deviceId, string hubName, string sasToken, int portNumber = 8883, string apiVersion = "2021-04-12", int wait = 5000)
```

Comment:
I understand the module has an MQTT broken embedded. So that's the reason why you're not using the Azure IoT library we have.
Still, can you align the API? So it's going to work the exact same way?
See: https://github.com/nanoframework/nanoFramework.Azure.Devices/blob/main/nanoFramework.Azure.Devices.Client/DeviceClient.cs

Reply:
And looking more at the code, I would split into 2 part the existing code:

* Keep what is generic in this file, so what's SIM7080 related in this file
* Move in another file what's related to Azure IoT, name it the same as in the Azure IoT library.

so in the long term, not for this PR: create interface to be able to use any MQTT library with the Azure IoT Libraray. So the Azure IoT library can be used in a more generic way.

Reply:
The support for MQTT provided by the SIM is added to the library; so that is native functionality and SIM7080 related?

I am not sure if I can combine the MQTT broker with the only the network connection. To be honest I didn't test that.
I will investigate that and also add authentication by certificate in the next release.